### PR TITLE
Multipliers become projection days for all fees related to BR

### LINKED ISSUE #822: Cumsum project over correct number of days
When we have a recommendation from red team and cryptoecon we will either 
1) always compute cumsum of ratio over 20 days and then scale it back depending on context (FF, SP, IP)
2) parameterize `ExpectedDayRewardForPower` by the days we are taking the cumsum so that we integrate over different periods for different calculations
3) keep it the same

fyi @zixuanzh @nicola @davidad

### PR DESCRIPTION
**Pull request by ZenGround0**
Closes #822 

Currently WIP because we aren't correctly storing one br and twenty br in state to compute termination fees.

### CODE REVIEW

File: actors/builtin/miner/monies.go [hunk 1]
```diff
@@ -9,36 +9,37 @@ import (
...
+var InitialPledgeProjectionPeriod = abi.ChainEpoch(InitialPledgeFactor) * builtin.EpochsInDay
 var LockTargetFactorNum = big.NewInt(3)
 var LockTargetFactorDenom = big.NewInt(10)
 
-// FF = (DeclaredFaultFactorNum / DeclaredFaultFactorDenom) * BR(t)
-var DeclaredFaultFactorNum = big.NewInt(214)
-var DeclaredFaultFactorDenom = big.NewInt(100)
+// FF = BR(t, DeclaredFaultProjectionPeriod)
+// projection period of 2.14 days  2880 * 2.14 = 6163.2.  Rounded to nearest epoch 6163
+var DeclaredFaultProjectionPeriod = abi.ChainEpoch(6163)
```
Comment (anorth) [inline actors/builtin/miner/monies.go:26]: Which epoch duration is this calculated with? Could you do the arithmetic with integers (e.g. scaled by 10000) so it stays current?

File: actors/builtin/miner/monies.go [hunk 2]
```diff
@@ -9,36 +9,37 @@ import (
...
+var UndeclaredFaultProjectionPeriod = abi.ChainEpoch(5) * builtin.EpochsInDay
 
 // Maximum number of days of BR a terminated sector can be penalized
 const TerminationLifetimeCap = abi.ChainEpoch(70)
 
 // This is the BR(t) value of the given sector for the current epoch.
 // It is the expected reward this sector would pay out over a one day period.
 // BR(t) = CurrEpochReward(t) * SectorQualityAdjustedPower * EpochsInDay / TotalNetworkQualityAdjustedPower(t)
-func ExpectedDayRewardForPower(rewardEstimate, networkQAPowerEstimate *smoothing.FilterEstimate, qaSectorPower abi.StoragePower) abi.TokenAmount {
+func ExpectedDayRewardForPower(rewardEstimate, networkQAPowerEstimate *smoothing.FilterEstimate, qaSectorPower abi.StoragePower, projectionDuration abi.ChainEpoch) abi.TokenAmount {
```
Comment (anorth) [inline actors/builtin/miner/monies.go:37]:
```suggestion
func ExpectedRewardForPower(rewardEstimate, networkQAPowerEstimate *smoothing.FilterEstimate, qaSectorPower abi.StoragePower, projectionDuration abi.ChainEpoch) abi.TokenAmount {
```

File: actors/builtin/miner/monies.go [hunk 3]
```diff
@@ -47,29 +48,26 @@ func ExpectedDayRewardForPower(rewardEstimate, networkQAPowerEstimate *smoothing
...
-		UndeclaredFaultFactorDenom)
+	_ = builtin.EpochsInDay
+	return ExpectedDayRewardForPower(rewardEstimate, networkQAPowerEstimate, qaSectorPower, UndeclaredFaultProjectionPeriod)
 }
 
 // Penalty to locked pledge collateral for the termination of a sector before scheduled expiry.
 // SectorAge is the time between the sector's activation and termination.
-func PledgePenaltyForTermination(dayRewardAtActivation abi.TokenAmount, sectorAge abi.ChainEpoch, rewardEstimate, networkQAPowerEstimate *smoothing.FilterEstimate, qaSectorPower abi.StoragePower) abi.TokenAmount {
+func PledgePenaltyForTermination(dayRewardAtActivation, twentyDayRewardAtActivation abi.TokenAmount, sectorAge abi.ChainEpoch, rewardEstimate, networkQAPowerEstimate *smoothing.FilterEstimate, qaSectorPower abi.StoragePower) abi.TokenAmount {
 	// max(SP(t), 20*BR(StartEpoch) + BR(StartEpoch)*min(SectorAgeInDays, 70))
```
Comment (anorth) [inline actors/builtin/miner/monies.go:64]:
```suggestion
	// max(SP(t), BR(StartEpoch, 20d) + BR(StartEpoch)*min(SectorAgeInDays, 70))
```

File: actors/builtin/miner/monies.go [hunk 4]
```diff
@@ -47,29 +50,26 @@ func ExpectedDayRewardForPower(rewardEstimate, networkQAPowerEstimate *smoothing
...
+	return ExpectedRewardForPower(rewardEstimate, networkQAPowerEstimate, qaSectorPower, DeclaredFaultProjectionPeriod)
 }
 
 // This is the SP(t) penalty for a newly faulty sector that has not been declared.
 // SP(t) = UndeclaredFaultFactor * BR(t)
 func PledgePenaltyForUndeclaredFault(rewardEstimate, networkQAPowerEstimate *smoothing.FilterEstimate, qaSectorPower abi.StoragePower) abi.TokenAmount {
-	return big.Div(
-		big.Mul(UndeclaredFaultFactorNum, ExpectedDayRewardForPower(rewardEstimate, networkQAPowerEstimate, qaSectorPower)),
-		UndeclaredFaultFactorDenom)
+	_ = builtin.EpochsInDay
```
Comment (anorth) [inline actors/builtin/miner/monies.go:59]: Stray

File: actors/builtin/miner/miner_state.go [hunk 5]
```diff
@@ -129,16 +129,17 @@ type SectorPreCommitOnChainInfo struct {
...
+	SealProof               abi.RegisteredSealProof // The seal proof type implies the PoSt proof/s
+	SealedCID               cid.Cid                 // CommR
+	DealIDs                 []abi.DealID
+	Activation              abi.ChainEpoch  // Epoch during which the sector proof was accepted
+	Expiration              abi.ChainEpoch  // Epoch during which the sector expires
+	DealWeight              abi.DealWeight  // Integral of active deals over sector lifetime
+	VerifiedDealWeight      abi.DealWeight  // Integral of active verified deals over sector lifetime
+	InitialPledge           abi.TokenAmount // Pledge collected to commit this sector
+	ExpectedDayReward       abi.TokenAmount // Expected one day projection of reward for sector computed at activation time
+	ExpectedTwentyDayReward abi.TokenAmount // Expected twenty day projection of reward for sector computed at activation time
```
Comment (davidad) [inline actors/builtin/miner/miner_state.go:142]: all things considered, def feel free to ignore this, but baking parameters like "twenty" into field names is not ideal

Comment (zixuanzh) [reply actors/builtin/miner/miner_state.go:142]: we can call this `InitialStoragePledge`

**Pull request merged**